lib/modules/collapseWhitespace.es6

```diff
@@ -7,8 +7,21 @@ const noWhitespaceCollapseElements = new Set([
     'textarea'
 ]);
 
+const noTrimWhitespacesArroundElements = new Set([
+    // non-empty tags that will maintain whitespace around them
+    'a', 'abbr', 'acronym', 'b', 'bdi', 'bdo', 'big', 'button', 'cite', 'code', 'del', 'dfn', 'em', 'font', 'i', 'ins', 'kbd', 'label', 'mark', 'math', 'nobr', 'object', 'q', 'rp', 'rt', 'rtc', 'ruby', 's', 'samp', 'select', 'small', 'span', 'strike', 'strong', 'sub', 'sup', 'svg', 'textarea', 'time', 'tt', 'u', 'var',
+    // self-closing tags that will maintain whitespace around them
+    'comment', 'img', 'input', 'wbr'
+]);
```

Comment:
Those tags will retain white spaces around them. If a "text node" is next to one of those tags, at least one white space has to remain.